api-service/src/services/DatasetService.ts

```diff
@@ -260,8 +280,9 @@ class DatasetService {
                 await this.createDruidDataSource(draftDataset, transaction);
             }
             if(indexingConfig.lakehouse_enabled) {
-                const liveDataset = await this.getDataset(draftDataset.dataset_id, ["id"], true);
-                if(liveDataset) {
+                const liveDataset = await this.getDataset(draftDataset.dataset_id, ["id", "api_version"], true);
+
+                if(liveDataset && liveDataset.api_version === "v2") {
```

Comment:
Why is the api_version === v2 check required here. If it is not v2 we would have rejected it during validation itself right?

Reply:
v1 datasets dosn't have the hudi spec to update, so if the dataset is v2 then only we are allowing it to update